src/scanner/ScannerUtils.spec.ts

```diff
@@ -110,5 +112,44 @@ describe('ScannerUtils', () => {
       expect(filteredPractices.practicesOff.length).toBeGreaterThanOrEqual(1);
       expect(filteredPractices.customApplicablePractices.length).toBeGreaterThanOrEqual(2);
     });
+
+    it('Filter correctly if practice impact is high and fail=high', () => {
+      const argumentsProvider = { uri: '.', fail: PracticeImpact.high };
+      const result = ScannerUtils.filterNotPracticingPracticesToFail(reportArguments, argumentsProvider);
+      expect(result).toHaveLength(1);
+    });
+
+    it('Filter correctly if practice impact is high and fail=off', () => {
+      const argumentsProvider = { uri: '.', fail: PracticeImpact.off };
+      const result = ScannerUtils.filterNotPracticingPracticesToFail(reportArguments, argumentsProvider);
+      expect(result).toHaveLength(0);
+    });
   });
 });
+
+const reportArguments: PracticeWithContextForReporter[] = [
```

Comment:
Remove this function and use `practiceWithContextFactory()` instead as used in `CLIReporter.spec.ts`